ADuCM360_demo_cn0409: Initial commit

mirceacaprioru opened the pull request:
The EVAL-CN0409-ARDZ shield is a water turbidity calculation application
that gives the NTU value of the water being tested. Is uses the ADPD105
chip which incorporates adc and led driver circuitry.

Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

stefpopa commented on projects/ADuCM360_demo_cn0409/src/main.c:
```diff
@@ -0,0 +1,203 @@
+	}
+
+	switch (status & COMIIR_STA_MSK) {                  /* Check what command to execute */
+	case COMIIR_STA_RXBUFFULL:                     /* Check if UART register is available to be read */
```
check that you don't cross 80 columns

stefpopa commented on projects/ADuCM360_demo_cn0409/src/main.c:
```diff
@@ -0,0 +1,203 @@
+
+		timer_sleep(3000);
+	}
+	// Infinite loop, never return.
```
decide for one type of comment either `//` or `/*`

rgetz replied:
Please use C style comments, I think that is more consistent with other projects we do, isn't it?

stefpopa commented on projects/ADuCM360_demo_cn0409/src/main.c:
```diff
@@ -0,0 +1,203 @@
+	CN0409_AFECalibration();
+	CN0409_ChannelOffsetCalibration();
+
+	// Infinite loopy
```
?

stefpopa commented on projects/ADuCM360_demo_cn0409/src/main.c:
```diff
@@ -0,0 +1,203 @@
+				uart_received = UART_TRUE;
+			}
+		}
+		if (c == 0x0D) {
```
use a define instead of "magic numbers"

stefpopa commented on projects/ADuCM360_demo_cn0409/src/main.c:
```diff
@@ -0,0 +1,203 @@
+	ui16I2cStatus = 0;
+	ui16I2cStatus = I2cSta(MASTER);
+
+	if((ui16I2cStatus & I2CMSTA_RXREQ) == I2CMSTA_RXREQ) { // Master Receive IRQ
```
you don't need brackets here

stefpopa commented on projects/ADuCM360_demo_cn0409/src/main.c:
```diff
@@ -0,0 +1,203 @@
+	AppPrintf("Test for turbidity\n\r");
+	AppPrintf("Device id for ADPD105: %04x\n\r", ADPD105_ReadReg(ADPD105_DEVID));
+
+	/*Init and configure CN0358*/
```
Add a blank space after` /* `

stefpopa commented on projects/ADuCM360_demo_cn0409/src/CN0409.c:
```diff
@@ -0,0 +1,745 @@
+float fpFirstPoint, fpSecondPoint;
+
+float y_code[3], x_ntu[3], xm, ym;
+float num, denom;
```
you can group more variables on one line

stefpopa commented on projects/ADuCM360_demo_cn0409/src/CN0409.c:
```diff
@@ -0,0 +1,745 @@
+float fpTurbidityRatio;
+float fpTurbidityNonRatio;
+
+uint32_t  u32Buffer[5];
```
the same as above

stefpopa commented on projects/ADuCM360_demo_cn0409/src/CN0409.c:
```diff
@@ -0,0 +1,745 @@
+	ADPD105_SelectLED(LEDX2, SLOTA); //configure led1 for SlotA
+	ADPD105_SelectLED(LEDX1, SLOTB); //configure led1 for SlotB
+
+	/*Set led course value*/
```
`/* Set led course value */ `-> add a blank space, please check this in the rest of the code as well

stefpopa commented on projects/ADuCM360_demo_cn0409/src/CN0409.c:
```diff
@@ -0,0 +1,745 @@
+	/*Set FIFO and start measurement*/
+	ADPD105_SetFIFO();
+	ADPD105_SetOperationMode(NORMAL_OPERATION); //set normal operation - start
+}
```
replace the "magic numbers" with defines or enums where possible

stefpopa commented on projects/ADuCM360_demo_cn0409/src/CN0409.c:
```diff
@@ -0,0 +1,745 @@
+	ADPD105_SetOffset(SLOTB, stOffset);
+
+	ADPD105_SetOperationMode(PROGRAM); //set programming mode
+	//Enable LEDs
```
please use type of comments consistently; check this everywhere in the code

stefpopa commented on projects/ADuCM360_demo_cn0409/src/ADPD105.c:
```diff
@@ -0,0 +1,487 @@
+	au16DataSlotB[0] = ADPD105_ReadReg(ADPD105_SLOTB_PD1_16BIT);
+	au16DataSlotB[1] = ADPD105_ReadReg(ADPD105_SLOTB_PD2_16BIT);
+	au16DataSlotB[2] = ADPD105_ReadReg(ADPD105_SLOTB_PD3_16BIT);
+	au16DataSlotB[3] = ADPD105_ReadReg(ADPD105_SLOTB_PD4_16BIT);
```
I don't like the fact that this function uses a global array; maybe pass a pointer to this array as a parameter to this function?

stefpopa commented on projects/ADuCM360_demo_cn0409/src/CN0409.c:
```diff
@@ -0,0 +1,745 @@
+	for (u8Count = 0; u8Count < 4; u8Count++) {
+		fpSlotACurrents[u8Count] = 0.0;
+		fpSlotBCurrents[u8Count] = 0.0;
+	}
```
you can probably use memset() instead of the for loop

stefpopa commented on projects/ADuCM360_demo_cn0409/src/CN0409.c:
```diff
@@ -0,0 +1,745 @@
+
+	float fp16DataSlotA[4] = {0,0,0,0};
+	float fp16DataSlotB[4] = {0,0,0,0};
+	float num_avg = 256;
```
why is this variable of type float?

stefpopa commented on projects/ADuCM360_demo_cn0409/include/CN0409.h:
```diff
@@ -0,0 +1,123 @@
+	RATIOMETRIC = 0,
+	NONRATIOMETRIC,
+	GLI
+} CN0409_Type;
```
we don't use typedefs anymore

stefpopa commented on projects/ADuCM360_demo_cn0409/src/ADPD105.c:
```diff
@@ -0,0 +1,487 @@
+**/
+void ADPD105_DisplayRegValues(void)
+{
+	//ADPD105_ReadDataRegs();
```
commented code

andrew-mclachlan commented on projects/ADuCM360_demo_cn0409/Debug/RTE/Device/ADuCM360/ClkLib.d:
```diff
@@ -0,0 +1,23 @@
+RTE/Device/ADuCM360/ClkLib.o: \
```
We should exclude (.gitignore) the Debug (and Release) folder as it's re-generated.

andrew-mclachlan replied:
I see that we've added Debug and Release to a .gitignore but it doesn't look like we've removed those intermediate directories from this Pull Request/commit.

mirceacaprioru commented:
Changelog V1->V2
- removed Debug folder and files
- changed commented code to /* */ style (C style comments)
- removed unused functions
- astyle linux formatting applied
- removed commented code
